### DISCUSSION TITLE
pyboard doesn't start my main.py

### DISCUSSION THREAD

**Discussion by joh02** [micropython/micropython#10447] [General]
Hello,
Beginner question: i can communicate with my pyboard v.1. via the pyboard.py tool. That is, run code and upload files etc. However, when I upload a main.py and press reset, I would expect the code inside to be executed, which is not the case. Better said, it seems to execute code of a previous version or that of another main (but which does not exist according to ls). The entry 
`
pyb.main('main.py')
`
in boot.py is active.
What can I do?
joh

**Comment by bixb922** [comment]
I would try with mpremote

**Comment by cve2022** [comment]
It seems to me you are not realy copying the script to your board. Instead, you are making it to run you main.py. From the [pyboard.py documentation](https://docs.micropython.org/en/latest/reference/pyboard.py.html):

## Running a script on the device

If you have a script, app.py that you want to run on a device, then use:

`$ pyboard.py --device /dev/ttyACM0 app.py`

Note that this doesn’t actually copy app.py to the device’s filesystem, it just loads the code into RAM and executes it. Any output generated by the program will be displayed.

**Comment by cve2022** [comment]
As suggested above, you could use mpremote, or rsync or even thonny to copy your scripts to your board. I suggest you use thonny: despite some issues, it is very convenient to use, especially for beginners like us. 😃

**Comment by steveh127** [comment]
I routinely use pyboard.py to copy files to a range of boards. The syntax you need is:

`pyboard.py --device /dev/ttyACM0 -f cp main.py`

Substituting your device name.

**Reply by jimmo** [reply]
That should be `pyboard.py --device /dev/ttyACM0 -f cp main.py :`  (include the destination path)

@joh02 In general using `mpremote` is preferred over pyboard.py as it supports all the functionality of pyboard.py, plus a lot of extra features and is easier to install (e.g. via pip).

**Reply by steveh127** [reply]
It was only inertia that had me using pyboard.py. Having had a look at mpremote it's significantly better. I use geany as my development environment and have menu items set up to execute code and copy files to the board. This worked fine with pyboard.py but switching from a /dev/ttyACM0 device to a /dev/ttyUSB0 board was a pain as I had to edit the commands. mpremote automatically detects connected boards of both types. And the syntax is simpler.

**Comment by joh02** [comment]
Thank you to everyone who responded to my question.
Especially for the reference to mpremote. What probably caused my issue is my linux mint distribution, sometimes the pyboard is mounted as mass storage, sometimes not. Same with the appearance of the /dev/ttyAS0 interface. All in all it takes some getting used to, but for now it works as suspected. 
Many thanks.
joh
